src/Microsoft.Azure.WebJobs.Logging/LogFactory.cs

```diff
@@ -15,23 +13,35 @@ public static class LogFactory
         /// <summary>
         /// Get a reader that reads from the given table. 
         /// </summary>
-        /// <param name="table"></param>
+        /// <param name="tableLookup"></param>
         /// <returns></returns>
-        public static ILogReader NewReader(CloudTable table)
+        public static ILogReader NewReader(ILogTableProvider tableLookup)
         {
-            return new LogReader(table);
+            return new LogReader(tableLookup);
         }
 
         /// <summary>
         /// Create a new writer for the given compute container name that writes to the given table.
         /// Multiple compute instances can write to the same table simultaneously without interference. 
         /// </summary>
         /// <param name="computerContainerName">name of the compute container. Likley %COMPUTERNAME%. </param>
-        /// <param name="table">underlying azure storage table to write to.</param>
+        /// <param name="tableLookup">underlying azure storage table to write to. 
+        /// Passed a string suffix (which will consist of valid Azure table characters). 
+        /// This must be a deterministic (replayable) function.</param>
         /// <returns></returns>
-        public static ILogWriter NewWriter(string computerContainerName, CloudTable table)
+        public static ILogWriter NewWriter(string computerContainerName, ILogTableProvider tableLookup)
+        {
+            return new LogWriter(computerContainerName, tableLookup);
+        }
+
+        /// <summary>
+        /// Get a default table provider for the given tableClient. This will generate table names with the given prefix.
+        /// </summary>
+        /// <param name="tableClient">storage client for where to generate tables</param>
+        /// <param name="tableNamePrefix">prefix for tables to generate. This should be a valid azure table name.</param>
+        public static ILogTableProvider NewTableProvider(CloudTableClient tableClient, string tableNamePrefix = LogFactory.DefaultLogTableName)
         {
-            return new LogWriter(computerContainerName, table);
+            return new EpochTableProvider(tableClient, tableNamePrefix);
```

Comment:
The dashboard will work with the default settings. That was part of the rational to keep the provider interface simple and not make it attractive to customize. But if a highly determined individual created their own arbitrary table provider here, it would log to it, but it wouldn't work with the dashboard (although they could deploy their own dashboard with their own custom logger) 

---
In reply to: [82910137](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82910137) [](ancestors = 82910137)